stablehlo/transforms/PassPipelines.cpp

```diff
@@ -28,6 +28,14 @@ void createStablehloDeserializePipeline(OpPassManager &pm) {
   pm.addPass(stablehlo::createVhloLegalizeToStablehloPass());
 }
 
+void createStablehloRefineArgumentsAndShapesPipeline(OpPassManager &pm,
```

Comment:
I feel like this name is a bit too specific, but also does not fully describe what the pipeline is for. To my understanding, the ultimate goal here is to remove dynamism. So maybe something like `createStablehloRemoveDynamismPipeline`? And the corresponding function in `Api.cpp` could be called `removeDynamism`?

Reply:
Thanks for the suggestion!

I agree with the point that the name is too specific and does not convey the overall goal. Thanks!

Given that the pass-pipeline can only remove dynamism in certain cases (e.g. as mentioned in the description there canse dynamic_ops* which cannot be removed just by using the pipeline), with that my impression is that the name would over-commit to what can be achieved.

What do you think about 
`createStablehloRefinePolymorphicModule` it has what it does?

Reply:
I think it's okay to "over-commit": most of our passes do it, including the passes that are included in this pipeline. "refine-shapes" can leave some shapes unrefined. "canonicalize-dynamism" can fail to remove some dynamic ops. More generally, passes are named after their intent, but they rarely promise anything, e.g. the CSE and DCE passes in MLIR do their best, but they can't possible handle everything (since it's undecidable anyway).

Ideally, we would have checks at the end of these passes such that if we fail to refine all shapes we fail the pass, and similarly if we fail to remove all dynamic ops we fail the pass. Currently, I believe dynamic shapes/ops can still make it to the interpreter, which will cause obtuse errors?

Reply:
> What do you think about
createStablehloRefinePolymorphicModule it has what it does?

I don't love the name "refine polymorphic module", because the term "polymorphism" is quite overloaded, and I feel like it doesn't really capture the intent of the pass. I don't think it's a huge deal, but IMO `removeDynamism` is a better name.

Reply:
Fair point. Thanks!
 
>> Ideally, we would have checks at the end of these passes such that if we fail to refine all shapes we fail the pass, and similarly if we fail to remove all dynamic ops we fail the pass. 

Agreed.

>>  Currently, I believe dynamic shapes/ops can still make it to the interpreter, which will cause obtuse errors?

That is right. With dynamic_ops it is op not supported by interpreter and dynamic_shapes which will fail on type mismatch with interpreterValue whle allocating resources for result type. 


>> I don't love the name "refine polymorphic module", because the term "polymorphism" i
I agree with some of it. The shape qualifier makes it less ambiguous. 

In any-case, I am fine with both option. Using `removeDyanmism` now.

Reply:
PTAL.